src/test/ui/type-alias-impl-trait/issue-57611-trait-alias.rs

```diff
@@ -0,0 +1,26 @@
+// Regression test for issue #57611
+// Ensures that we don't ICE
+
+#![feature(trait_alias)]
+#![feature(type_alias_impl_trait)]
+
+trait Foo {
+    type Bar: Baz<Self, Self>;
+
+    fn bar(&self) -> Self::Bar;
+}
+
+struct X;
+
+impl Foo for X {
+    type Bar = impl Baz<Self, Self>; //~ ERROR type mismatch in closure arguments
+    //~^ ERROR type mismatch resolving
+
+    fn bar(&self) -> Self::Bar {
+        |x| x
+    }
+}
+
+trait Baz<A, B> = Fn(&A) -> &B;
```

Comment:
Hmm... It seems like this should be check-pass? If I substitute uses of the trait alias for its body (`Fn(&Self) -> &Self`) in each location the test does pass. (I think we should still land this PR, but I would like a FIXME noting the question.)